superset/db_engine_specs/presto.py

```diff
@@ -906,21 +906,23 @@ def extra_table_metadata(
         indexes = database.get_indexes(table_name, schema_name)
         if indexes:
             cols = indexes[0].get("column_names", [])
-            full_table_name = table_name
-            if schema_name and "." not in table_name:
-                full_table_name = "{}.{}".format(schema_name, table_name)
-            pql = cls._partition_query(full_table_name, database)
-            col_names, latest_parts = cls.latest_partition(
-                table_name, schema_name, database, show_first=True
-            )
+            
+            if cols:
+                full_table_name = table_name
```

Comment:
Same logic as previous just indented under the `if cols:` statement. Unlike Presto where `get_indexes` returns `[]` for a non-partition table, Trino returns `[{'name': 'partition', 'column_names': [], 'unique': False}]`. Rather than overriding the engine specific `normalize_indexes` method I though it would be more prudent to make this method more robust given there was already an expectation that there may be no columns associated with the index, i.e., a non-partitioned table.